titiler/api/deps.py

```diff
@@ -230,3 +232,21 @@ def __init__(
         kwargs.pop("bounds", None)
         kwargs.pop("assets", None)  # For STAC
         self.kwargs = kwargs
+
+
+class CommonMosaicParams:
+    """Common mosaic params."""
+
+    def __init__(
+        self,
+        mosaic_id: Optional[str] = Query(None, description="MosaicJSON ID"),
+        url: Optional[str] = Query(None, description="MosaicJSON URL"),
+    ):
+        """Create mosaic path from args"""
+        if not mosaic_id and not url:
+            raise HTTPException(
+                status_code=422, detail="Missing 'mosaic_id' or 'url' parameter"
+            )
+        self.mosaic_path = (
+            url or f"{DEFAULT_MOSAIC_BACKEND}{DEFAULT_MOSAIC_HOST}/{mosaic_id}.json.gz"
+        )
```

Comment:
so, if we don't use mosaicid in the path maybe we can use a url like parameter like `mosaicid://{mosaicid}`?

Reply:
Thats a fantastic idea

Reply:
https://github.com/developmentseed/titiler/pull/40/commits/37f87415d21994e853230fb95f0e18771f8cbadc